src/networkd.c

```diff
@@ -314,6 +314,8 @@ write_network_file(net_definition* def, const char* rootdir, const char* path)
         g_string_append(s, "DHCP=ipv4\n");
     else if (def->dhcp6)
         g_string_append(s, "DHCP=ipv6\n");
+    if (!def->use_dns)
+        g_string_append(s, "UseDNS=false\n");
 
```

Comment:
Here you check for `!def->use_dns`, but I can't see anywhere where you'd set the default to be true. So I think this would change behaviour?

Reply:
Oh, and please could you add code to nm.c that throws an error or at least a warning if the value is false for an interface rendered by NM?

Reply:
I'll use `ignore-auto-dns` per https://developer.gnome.org/NetworkManager/stable/nm-settings.html